dashboard/lib/service/appengine_cocoon.dart

```diff
@@ -67,6 +70,38 @@ class AppEngineCocoonService implements CocoonService {
     }
   }
 
+  @override
+  Future<CocoonResponse<List<CommitTasksStatus>>> fetchCommitStatusesFirestore({
+    CommitStatus? lastCommitStatus,
+    String? branch,
+    required String repo,
+  }) async {
+    final Map<String, String?> queryParameters = <String, String?>{
+      if (lastCommitStatus != null) 'lastCommitKey': lastCommitStatus.commit.key.child.name,
+      'branch': branch ?? _defaultBranch,
+      'repo': repo,
+    };
+    final Uri getStatusUrl = apiEndpoint('/api/public/get-status-firestore', queryParameters: queryParameters);
+
+    /// This endpoint returns JSON [List<CommitTasksStatus>]
+    final http.Response response = await _client.get(getStatusUrl);
+
+    if (response.statusCode != HttpStatus.ok) {
+      return CocoonResponse<List<CommitTasksStatus>>.error(
+        '/api/public/get-status-firestore returned ${response.statusCode}',
+      );
+    }
+
+    try {
+      final Map<String, dynamic> jsonResponse = jsonDecode(response.body);
+      return CocoonResponse<List<CommitTasksStatus>>.data(
+        _commitStatusesFromJsonFirestore(jsonResponse['Statuses']),
+      );
+    } catch (error) {
```

Comment:
Is there a specific error we can catch?

Reply:
I am not sure about potential error types here. At this stage, I think it would be good to keep logics the same as before. We can identify different error types and do different actions when everything sets down. Does this sound good to you?

Reply:
That works for me.